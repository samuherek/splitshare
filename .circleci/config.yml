version: 2
jobs:
  build:
    working_directory: ~/splitshare
    docker:
      - image: circleci/node:chakracore-10.13
        environment:
          NODE_ENV: production
          DATABASE_URL: postgresql://ubuntu@localhost/splitshare-test?sslmode=disable
      - image: circleci/postgres:9.6.9
        environment:
          POSTGRES_USER: ubuntu
          POSTGRES_DB: splitshare-test
          POSTGRES_PASSOWRD: ''
    steps:
      - checkout
      - restore_cache:
          key: deps1-{{ .Branch }}-{{ checksum "package.json" }}
      - run:
          name: Instsall dependencies
          command: yarn install
      - save_cache:
          key: deps1-{{ .Branc }}-{{ checksum "package.json" }}
          paths:
            - ./node_modules

workflows:
  version: 2
  build:
    jobs:
      - build
