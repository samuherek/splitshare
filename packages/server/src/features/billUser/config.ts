export const inviteState = {
  PENDING: 'PENDING',
  REJECTED: 'REJECTED',
  ACCEPTED: 'ACCEPTED',
};
