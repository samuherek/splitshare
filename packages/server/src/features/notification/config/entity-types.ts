export type EntityTypeId = number;
export type ActionTypeId = number;
export type IdType = number;

export const ENTITY_TYPE = {
  BILL: 1,
  BILL_INVITE: 2,
  RECEIPT: 3,
};

export const ACTION_TYPE = {
  CREATED: 1,
  UPDATED: 2,
  DELETED: 3,
  ARCHIVED: 4,
};

export const NotificationUtils = {
  getEntity: (id: number) => {
    const entry = Object.entries(ENTITY_TYPE).find(
      ([_, value]) => value === id
    );

    if (!entry) {
      throw new Error('Could not find such notification entity');
    }

    return entry[0];
  },
  getAction: (id: number) => {
    const entry = Object.entries(ACTION_TYPE).find(
      ([_, value]) => value === id
    );

    if (!entry) {
      throw new Error('Could not find such notification action');
    }

    return entry[0];
  },
  sameType: (id: number, compare: number) => {
    return id === compare;
  },
};
