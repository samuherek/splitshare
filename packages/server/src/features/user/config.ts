export const userState = {
  ONBOARDING_VERIFY_EMAIL: 'ONBOARDING_VERIFY_EMAIL',
  ONBOARDING_PROFILE: 'ONBOARDING_PROFILE',
  ACTIVE: 'ACTIVE',
};
