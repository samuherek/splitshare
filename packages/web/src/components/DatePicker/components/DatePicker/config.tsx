import { WithViewProps } from '../../typings/SharedPickerProps';

export interface DatePickerProps
  extends WithViewProps<'year' | 'date' | 'month'> {}
