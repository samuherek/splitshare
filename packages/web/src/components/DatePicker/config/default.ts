export const datePickerDefaultProps = {
  minDate: new Date('1900-01-01'),
  maxDate: new Date('2100-01-01'),
  invalidDateMessage: 'Invalid Date Format',
  minDateMessage: 'Date should not be before minimal date',
  maxDateMessage: 'Date should not be after maximal date',
};
