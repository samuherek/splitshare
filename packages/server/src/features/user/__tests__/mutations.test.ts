import gql from 'graphql-tag';
import createTestServer from '../../../test-utils/server';
import { UserState } from '../config';

const baseUser = {
  email: 'email@email.com',
  firstName: null,
  lastName: null,
  avatarUrl: null,
  state: UserState.ACTIVE,
};

describe('user mutation', () => {
  describe('setupAccount', () => {
    it('should match snapshot', async () => {
      const SETUP_ACCOUNT = gql`
        mutation SetupAccount($input: SetupInput!) {
          setupAccount(input: $input) {
            id
            email
            firstName
            lastName
            avatarUrl
            state
          }
        }
      `;

      const { mutate } = await createTestServer({
        user: { id: 1 },
        models: {
          User: {
            update: jest.fn(() => ({
              ...baseUser,
              id: 1,
            })),
          },
        },
      });

      const res = await mutate({
        mutation: SETUP_ACCOUNT,
        variables: { input: { firstName: 'sam', lastName: 'uherek' } },
      });

      expect(res).toMatchSnapshot();
    });
  });
});
