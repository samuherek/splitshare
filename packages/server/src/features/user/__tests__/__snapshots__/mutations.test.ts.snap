// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user mutation setupAccount should match snapshot 1`] = `
Object {
  "data": Object {
    "setupAccount": Object {
      "avatarUrl": null,
      "email": "email@email.com",
      "firstName": null,
      "id": "1",
      "lastName": null,
      "state": "ACTIVE",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
